sycl/source/detail/graph_impl.hpp

```diff
@@ -606,7 +612,8 @@ class node_impl {
                      sycl::detail::kernel_param_kind_t::kind_invalid) {
             Type = "Invalid";
           }
-          Stream << i << ") Type: " << Type << " Ptr: " << Arg.MPtr << "\\n";
+          Stream << i << ") Type: " << Type << " Ptr: 0x" << std::hex << Ptr
```

Comment:
I'm not sure if `Ptr` will be valid for any of the non `kernel_param_kind_t::kind_pointer` types either, if `Arg.MPtr` is a `void*` that needs dereferenced to show any information that's interesting to the user.

Would suggest that we only print this pointer info for USM arguments, e.g.:

```cpp
Stream << i << ") Type: " << Type;
if (Arg.MType == sycl::detail::kernel_param_kind_t::kind_pointer) {
  Stream << " Ptr: 0x" << std::hex << Ptr << std::dec;
}
Stream << "\\n";
```

Reply:
I also have the concern on the case of the non kernel_param_kind_t::kind_pointer types, and don't have the answer.

My idea is to keep the `non kernel_param_kind_t::kind_pointer types` unchanged just in case that it helps.  I just found that the Arg.MPtr for kernel_param_kind_t::kind_pointer matches the value of level zero API trace with unitrace.

Let me try to change the code in another method.